### PR TITLE
SDK: AppOptions -> AppState

### PR DESCRIPTION
**Pull request by rigelrozanski**

### CODE REVIEW

File: types/genesis.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type GenesisDoc struct {
 	ConsensusParams *ConsensusParams   `json:"consensus_params,omitempty"`
 	Validators      []GenesisValidator `json:"validators"`
 	AppHash         cmn.HexBytes       `json:"app_hash"`
-	AppOptions      interface{}        `json:"app_options,omitempty"`
+	AppStateJSON    json.RawMessage    `json:"app_options,omitempty"`
```
Comment (ebuchman) [inline types/genesis.go]: We should be consistent with the Go name and JSON name

**Comment by rigelrozanski** [comment]
@ebuchman I forced pushed to (json desc now matches)

File: types/genesis.go (hunk 1 above)
Comment (rigelrozanski) [reply types/genesis.go]: k, Also just renamed to `AppState` unless there is a reason to include JSON in the name? Also what's with `omitempty`? I left it in there

**Comment by melekes** [comment]
What tutorials/docs need to be updated?

**Comment by ebuchman** [comment]
This is a breaking change to the genesis file, though it's not tested. 

We should
1) add a test to types/genesis_test.go to test that we get the right raw JSON for the app_state
2) decide if we want to include it in 0.16, and need to include in changelog

**Comment by greg-szabo** [comment]
Also a note from the PM: if we can avoid it, let's not move the goalpost too much. If we keep adding new issues to the sprint, we'll never finish it.
There is no description to the PR.
There is no issue associated with it.
I literally can't estimate anything about this PR. It's out of the blue.

Are we delivering a broken 0.16 codebase if we omit this PR from it and move it to the next release?
If the answer is no, let's move this to "next release".

**Comment by ebuchman** [comment]
This is coming from recent work on the SDK to just improve the name of the field to be what it actually is :). 

No we don't need it for this release

**Comment by melekes** [comment]
Todo:

- [ ] update changelog
- [ ] check/update documentation

**Pull request merged**